onnx/test/automatic_downgrade_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) ONNX Project Contributors
+
+# SPDX-License-Identifier: Apache-2.0
+import unittest
+
+import automatic_conversion_test_base
+import numpy as np
+
+import onnx
+from onnx import helper
+
+#####################################################################################
+# Every test calls _test_op_conversion to downgrade a model from the most recent opset version
+# to a early version and runs checker + shape inference on the downgraded model.
+####################################################################################
+
+
+class TestAutomaticDowngrade(automatic_conversion_test_base.TestAutomaticConversion):
+    def _test_op_downgrade(self, op, *args, **kwargs):
```

Comment:
```suggestion
    def _test_op_downgrade(self, op: str, *args, **kwargs):
```

I would annotate types for all new functions